docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,199 @@
+User Customization & Persistence
+================================
+
+Context
+-------
+
+.. todo::
+
+   This section will be handled by the requirements PR.
+
+
+Design Choices
+--------------
+
+:term:`ConfigMap` is chosen as a unified data access and storage media for user
+editable configurations in a MetalK8s cluster based on the above requirements
+for the following reasons:
+
+* Ability to support Update operations on ConfigMap's with CLI and UI easily
+  using our already existing python kubernetes module.
+* Guarantee of adaptability and ease of changing the design and implementation
+  in cases were customer needs evolve rapidly.
+* ConfigMaps stored in the :term:`etcd` database which is generally being
+  backed up. This ensures that user settings cannot be lost easily.
+
+
+How it works
+~~~~~~~~~~~~
+
+During Bootstrap, pillar values for the various services will be pre-filled
+with default values. Service pods and deployments will be configured to either
+consume configuration data directly from their ConfigMap or fallback to
+the default pillar values if the service specific ConfigMap was not found.
+
+**Using Salt states**
+
+Once a ConfigMap is updated by the user (say a user changes the number of
+replicas for Prometheus deployments to a new value), then perform the
+following actions:
+
+  - Apply a Salt states that reads the ConfigMap object, validates the schema
+    and checks the new values passed.
+
+  - Finally, we make sure that the updated values are picked up by their
+    respective consumers (this might require Pod restarts for changes to take
+    effect).
```

Comment:
The fact that Pods may be restarted is always the case in K8s, not sure this should be mentioned. In the end, Salt (MetalK8s) is not restarting any Pods. This should all be managed by the `Deployment`/`StatefulSet`/`DaemonSet` controllers because Salt updates some high-level object(s).

Reply:
removed